### PR TITLE
bump highs commit

### PR DESCRIPTION
**Pull request by matbesancon**
Set HiGHS commit to latest. We will try to see if windows works and if the previous fixes are still necessary
ping @galabovaa

### CODE REVIEW

File: H/HiGHS/build_tarballs.jl [hunk 1]
```diff
@@ -3,11 +3,11 @@
 using BinaryBuilder, Pkg
 
 name = "HiGHS"
-version = v"0.1.4"
+version = v"0.1.5"
 
 # Collection of sources required to complete build
 sources = [
-    GitSource("https://github.com/ERGO-Code/HiGHS.git", "0b4cb7f882a6c5d208fbb2edb4d8c416291ed4f5"),
+    GitSource("https://github.com/ERGO-Code/HiGHS.git", "2539ade9b0799f4ebd35c0ffe7c33996298cf363"),
```
Comment (giordano) [inline H/HiGHS/build_tarballs.jl:10]: The revision hash looks wrong

Comment (matbesancon) [reply H/HiGHS/build_tarballs.jl]: yes indeed, I think I took it from the commit inside the latest PR instead of latest commit

**Comment by matbesancon** [comment]
I updated the PR, how come github does not update the PR here?

**Comment by matbesancon** [comment]
Error is:
```
[16:51:30]   ninja (virtual):
[16:51:30]     provided by: samurai
[16:51:30]     required by: world[ninja]
```

**Comment by giordano** [comment]
No, the problem is that `apk` is momentarily malfunctioning.  You'd need to run either
```
apk del ninja
```
or
```
apk add samurai
```
before running other `apk` commands

**Comment by matbesancon** [comment]
Thanks! Do you have info on the bug? To know if we should add the fix to the script or wait for it to be fixed upstream before continuing?

**Comment by giordano** [comment]
#2356

**Comment by matbesancon** [comment]
It seems the current version is not backward-compatible with libgfortran3

**Comment by giordano** [comment]
> It seems the current version is not backward-compatible with libgfortran3

More than "libgfortran3", the problem might be that the code doesn't compile with older compilers.  It's failing in C++ source files

**Comment by galabovaa** [comment]
I managed to build locally an updated jll. I tried to open a PR here a couple of weeks ago but got permissions error. Do I need to have extra permissions to open a PR?

The libgfortran part I have removed. The fortran is necessary only for the fortran interface and I have customised CMake to remove it when building for Julia. The updated build_tarballs.jl I have contains the modifications, updated CMake settings and the commit I used to build successfully. 

Alternatively I can edit the build_tarballs file here from the web interface of GitHub and that should help me avoid the permissions error I got before, temporarily. I would still like to be able to open PRs, though.

**Comment by giordano** [comment]
> I tried to open a PR here a couple of weeks ago but got permissions error. Do I need to have extra permissions to open a PR?

No, you may be having problems with GitHub authentication: the method used by the latest stable version of `BinaryBuilder` is not working anymore.  We have a fix in the development version, which however requires the development version of Julia.  At the moment, the easiest thing to do is to deploy to file the build script generated by the wizard and manually open the PR

**Comment by galabovaa** [comment]
That would be it. I wish I had asked that question sooner. Can you tell me in more detail what you mean by "eploy to file the build script generated by the wizard and manually open the PR"? 

I have not used the wizard but only the command line interface with --deploy=local setting to generate the files locally. I tried to run the wizard but got an error and @matbesancon suggested it was intended for first time generation of jll packages?

@matbesancon how did you get the GitHub authentication to work? Do you have 2-factor authentication?

**Comment by giordano** [comment]
> That would be it. I wish I had asked that question sooner. Can you tell me in more detail what you mean by "eploy to file the build script generated by the wizard and manually open the PR"?

Oh, sorry, I assumed you had permissions problems with the automatic opening of PRs of the wizard.  Then I'm not sure what kind of permission error you may have run into :confused: 

> I have not used the wizard but only the command line interface with --deploy=local setting to generate the files locally.

Yeah, that's totally fine, that's what I usually do.

**Comment by matbesancon** [comment]
> More than "libgfortran3", the problem might be that the code doesn't compile with older compilers. It's failing in C++ source files

noob question, how come it succeeds with cxx03 then? I Shouldn't the Cxx standard fail in any case if it's a C++ compiler version?

**Comment by giordano** [comment]
"cxx03" isn't the C++ standard, but the ABI used by GCC to represent `std::string`, which they changed when they had to implement the C++ standard (that's why the old ABI is referred to as "C++03" and the new one as "C++11"): https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_dual_abi.html.

You can build with GCC a program with C++03 standard and C++11-style string ABI, and a program with C++11 standard and C++03-style string ABI (this latter case is the current one).

**Comment by matbesancon** [comment]
@galabovaa what are the minimum compiler requirements for HiGHS? The current v4.9 does not seem to be enough

**Comment by matbesancon** [comment]
The readme still says 4.9 minimum

**Comment by matbesancon** [comment]
Also for windows failing, is there some special instruction to follow to make it work?

**Comment by giordano** [comment]
```
[18:33:06] cd /workspace/srcdir/HiGHS/build/check && /opt/bin/x86_64-w64-mingw32-libgfortran5-cxx11/x86_64-w64-mingw32-g++ --sysroot=/opt/x86_64-w64-mingw32/x86_64-w64-mingw32/sys-root/  @CMakeFiles/unit_tests.dir/includes_CXX.rsp -Wall -Wextra -Wno-unused-parameter -pedantic -O3 -DNDEBUG -fopenmp -std=gnu++11 -o CMakeFiles/unit_tests.dir/TestIO.cpp.obj -c /workspace/srcdir/HiGHS/check/TestIO.cpp
[18:33:07] In file included from /workspace/srcdir/HiGHS/check/TestIO.cpp:5:
[18:33:07] /workspace/srcdir/HiGHS/external/catch.hpp: In instantiation of ‘bool Catch::compareNotEqual(const LhsT&, RhsT&&) [with LhsT = char*; RhsT = const long long int&]’:
[18:33:07] /workspace/srcdir/HiGHS/external/catch.hpp:1560:37:   required from ‘const Catch::BinaryExpr<LhsT, const RhsT&> Catch::ExprLhs<LhsT>::operator!=(const RhsT&) [with RhsT = long long int; LhsT = char* const&]’
[18:33:07] /workspace/srcdir/HiGHS/check/TestIO.cpp:60:5:   required from here
[18:33:07] /workspace/srcdir/HiGHS/external/catch.hpp:1534:96: error: ISO C++ forbids comparison between pointer and integer [-fpermissive]
[18:33:07]      auto compareNotEqual( LhsT const& lhs, RhsT&& rhs ) -> bool { return static_cast<bool>(lhs != rhs); }
[18:33:07]                                                                                             ~~~~^~~~~~
[18:33:07] make[2]: *** [check/CMakeFiles/unit_tests.dir/build.make:111: check/CMakeFiles/unit_tests.dir/TestIO.cpp.obj] Error 1
[18:33:07] make[2]: Leaving directory '/workspace/srcdir/HiGHS/build'
[18:33:07] make[1]: *** [CMakeFiles/Makefile2:255: check/CMakeFiles/unit_tests.dir/all] Error 2
[18:33:07] make[1]: Leaving directory '/workspace/srcdir/HiGHS/build'
[18:33:07] make: *** [Makefile:182: all] Error 2
```
Look like plain errors in the source code?

**Comment by matbesancon** [comment]
These look a lot like the ones I noticed the first time:
https://github.com/ERGO-Code/HiGHS/issues/311#issuecomment-613053920

**Comment by giordano** [comment]
The error in `catch` happens during `make install`, which is quite interesting...  Looks like something in the CMake scripts is messed up

**Comment by matbesancon** [comment]
> Edit: but this is in catch, can you disable the tests?

there is a fast build option but it does not build IPX, which could be interesting to keep (I think?)

File: H/HiGHS/build_tarballs.jl [hunk 2]
```diff
@@ -23,6 +23,7 @@ if [[ "${target}" == *86*-linux-musl* ]]; then
 fi
 mkdir -p HiGHS/build
 cd HiGHS/build
+apk del ninja
```
Comment (fingolfin) [inline H/HiGHS/build_tarballs.jl:26]: This should not be needed anymore thanks to PR #2356
```suggestion
```

**Comment by fingolfin** [comment]
There seem to be two major sets of failures here:
1. Windows builds fail. Of course you could just disable them. But I think there is a chance that it is something simple; like, it seems as if the DLLs are put into the `lib` dir (as usual on Unix), but on Windows they need to be moved to `bin`; multiple recipes already do that, you can search for that; something like this at the end of your build recipe may do it:
   ```sh
   if [[ ${target} == *mingw* ]]; then
       mv bin/*.dll ${libdir}
   fi
   ```

2. failures for the libgfortran3 variant. Simple solution: drop those target platforms. Alternatively: try to work around it. The compile error I see is that `size_t` is not defined; that suggests a missing `#include` which should be easy to patch (if one knows a bit of C/C++), but of course it might just be the tip of the iceberg...

**Comment by matbesancon** [comment]
thanks for the tips. For Windows I'll check update the build.

> failures for the libgfortran3 variant

is there a way to condition the libgfortran variant? With a compiler version?

**Comment by matbesancon** [comment]
@galabovaa I'm not sure what the HiGHS team would prefer in the options above, opinion welcome :)

**Pull request closed without merging**